modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/ObjectEntryOpenAPIContributor.java

```diff
@@ -789,23 +785,24 @@ else if (!pathName.contains("/permissions")) {
 		}
 	}
 
-	private void _setReadOnlyProperties(OpenAPI openAPI) {
+	private void _setReadOnlyProperties(Map<String, Schema> schemas) {
 		Map<String, ObjectField> objectFields =
 			ObjectFieldUtil.toObjectFieldsMap(
 				_objectFieldLocalService.getObjectFields(
 					_objectDefinition.getObjectDefinitionId()));
 
-		Schema schema = _getObjectDefinitionSchema(openAPI);
+		Schema objectDefinitionSchema = schemas.get(
+			_objectDefinition.getShortName());
 
-		Map<String, Schema> properties = schema.getProperties();
+		Map<String, Schema> properties = objectDefinitionSchema.getProperties();
 
 		for (Map.Entry<String, Schema> entry : properties.entrySet()) {
 			String key = entry.getKey();
 
-			schema = entry.getValue();
+			objectDefinitionSchema = entry.getValue();
```

Comment:
This renaming is weird, as the first time it is the ObjectDefinition schema, but in the following ones, it is taking the value of every property schema

Reply:
Why don't you just leave it as `schema` instead? orusing `objectDefinitionSchema` and here `schema`